server/test_test.go

```diff
@@ -112,17 +111,17 @@ func require_Error(t *testing.T, err error, expected ...error) {
 	t.Fatalf("Expected one of %v, got '%v'", expected, err)
 }
 
-func require_Equal(t *testing.T, a, b string) {
+func require_Equal[T comparable](t *testing.T, a, b T) {
 	t.Helper()
-	if strings.Compare(a, b) != 0 {
-		t.Fatalf("require equal, but got: %v != %v", a, b)
+	if a != b {
+		t.Fatalf("require %T equal, but got: %v != %v", a, a, b)
 	}
 }
 
-func require_NotEqual(t *testing.T, a, b [32]byte) {
+func require_NotEqual[T comparable](t *testing.T, a, b T) {
 	t.Helper()
-	if bytes.Equal(a[:], b[:]) {
-		t.Fatalf("require not equal, but got: %v != %v", a, b)
+	if a == b {
```

Comment:
Will this do the right thing for 2 different byte slices that have same contents?

Reply:
The function was previously taking `a, b [32]byte` and fixed-size arrays can be compared using `==`, so it will continue to work for what it was being used for.

Trying to pass in slices instead of fixed-size arrays will result in a compile-time error as they are not `comparable`.